test/unit/test_annotated_code.c

```diff
@@ -0,0 +1,305 @@
+	}
+	if (first->type == R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT) {
+		mu_assert_eq (first->syntax_highlight.type, second->syntax_highlight.type, "syntax highlight type of offset doesn't match");
+	}
+}
+
+static bool test_r_annotated_code_add_annotation() {
+	char *test_string = malloc (54 * sizeof (char));
+	strcpy (test_string, "abcdefghijklmnopqrtstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+	RVector /*<RCodeAnnotation>*/ *test_annotations;
+	test_annotations = get_some_code_annotation_for_add ();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (test_annotations, annotation) {
+		r_annotated_code_add_annotation (code, annotation);
+	}
+
+	//Comparing
+	test_equal (r_vector_index_ptr (&code->annotations, 0), r_vector_index_ptr (test_annotations, 0));
+	test_equal (r_vector_index_ptr (&code->annotations, 1), r_vector_index_ptr (test_annotations, 1));
+
+	mu_end;
+}
+
+static bool test_r_annotated_code_annotations_in() {
+	char *test_string = malloc (54 * sizeof (char));
+	strcpy (test_string, "abcdefghijklmnopqrtstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+	RVector /*<RCodeAnnotation>*/ *test_annotations;
+	test_annotations = get_some_annotations_for_in ();
+
+	RCodeAnnotation *annotation;
+	r_vector_foreach (test_annotations, annotation) {
+		r_annotated_code_add_annotation (code, annotation);
+	}
+
+	RPVector *out = r_annotated_code_annotations_in (code, 11);
+	//Expecting indices = 3, 4, 5
+	mu_assert_eq (out->v.len, 3, "Additional annotations found. Bad output.");
+
+	test_equal (*r_pvector_index_ptr (out, 0), r_vector_index_ptr (test_annotations, 3));
+	test_equal (*r_pvector_index_ptr (out, 1), r_vector_index_ptr (test_annotations, 4));
+	test_equal (*r_pvector_index_ptr (out, 2), r_vector_index_ptr (test_annotations, 5));
+
+	mu_end;
+}
+
+static bool test_r_annotated_code_annotations_range() {
+	char *test_string = malloc (54 * sizeof (char));
+	strcpy (test_string, "abcdefghijklmnopqrtstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+	RAnnotatedCode *code = r_annotated_code_new (test_string);
+	RVector /*<RCodeAnnotation>*/ *test_annotations;
+	test_annotations = get_some_annotations_for_in ();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (test_annotations, annotation) {
+		r_annotated_code_add_annotation (code, annotation);
+	}
+
+	RPVector *out = r_annotated_code_annotations_range (code, 7, 16);
+	// Expecting indices = 2, 3, 4, 5
+	mu_assert_eq (out->v.len, 4, "Additional annotations found. Bad output.");
+
+	test_equal (*r_pvector_index_ptr (out, 0), r_vector_index_ptr (test_annotations, 2));
+	test_equal (*r_pvector_index_ptr (out, 1), r_vector_index_ptr (test_annotations, 3));
+	test_equal (*r_pvector_index_ptr (out, 2), r_vector_index_ptr (test_annotations, 4));
+	test_equal (*r_pvector_index_ptr (out, 3), r_vector_index_ptr (test_annotations, 5));
+	mu_end;
```

Comment:
Leak of code and out:
```suggestion

	r_pvector_free (out);
	r_annotated_code_free (code);
	mu_end;
```